alchemist/alchemist-euclidean-geometry/src/main/kotlin/it/unibo/alchemist/model/interfaces/geometry/euclidean2d/Intersection2D.kt

```diff
@@ -17,24 +17,38 @@ import it.unibo.alchemist.model.interfaces.geometry.Vector2D
  * The requirement is not explicitly enforced to allow the class to work covariantly.
  */
 sealed class Intersection2D<out V> {
+    /**
+     * List of intersection points (in case of infinite points this is empty).
+     */
+    open val asList: List<V> = emptyList()
     /**
      * Objects do not intersect.
      */
     object None : Intersection2D<Nothing>()
     /**
      * Objects intersect in a single [point].
      */
-    data class SinglePoint<P : Vector2D<P>>(val point: P) : Intersection2D<P>()
-    /**
-     * Objects intersect in multiple [points].
-     */
-    data class MultiplePoints<P : Vector2D<P>>(val points: List<P>) : Intersection2D<P>()
+    data class SinglePoint<P : Vector2D<P>>(val point: P) : Intersection2D<P>() {
```

Comment:
These can't be extended because they're data classes, and even if they weren't, if `SinglePoint` extended `MultiplePoints` it would have both `point` and `points` properties.. not very intuitive to me.

Reply:
Well, the solution'd be easy, you just need them not to be `data class`es.
The two properties would not be an issue IHMO, however there's a very practical problem I have not considered: type check on the sealed type would become order-dependent and counterintuitive, as `is MultiplePoints` would match for `SinglePoint`

Reply:
> type check on the sealed type would become order-dependent and counterintuitive, as `is MultiplePoints` would match for `SinglePoint`

Ah that would break a lot of code. I'll mark this as solved.